test/api/stripe_test.go

```diff
@@ -147,6 +151,134 @@ func testStripeCheckoutPOST(t *testing.T, at *test.AccountsTester) {
+	// running off different databases.
+	// See https://linear.app/skynetlabs/issue/SKY-1239/accounts-tester-parallel-testers
+	_, _, err = at.UserPUT("", "", stripeID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer gock.Off()
+	// We need to enable networking in order to allow the Tester to call our
+	// own API.
+	gock.EnableNetworking()
+	// Set up a response that will upgrade the user to tier 20.
+	gock.New("https://api.stripe.com").
+		Get("/v1/checkout/sessions/" + sessionIDWithSub20).
+		Reply(http.StatusOK).
+		Body(strings.NewReader(fixtures.StripeCheckoutSessionWithSubTier20))
+	// Set up a response that won't upgrade a tier 20 user because it's tier 5.
+	gock.New("https://api.stripe.com").
+		Get("/v1/checkout/sessions/" + sessionIDWithSub5).
+		Reply(http.StatusOK).
+		Body(strings.NewReader(fixtures.StripeCheckoutSessionWithSubTier5))
+	// Set up a response without a subscription.
+	gock.New("https://api.stripe.com").
+		Get("/v1/checkout/sessions/" + sessionIDWithoutSub).
+		Reply(http.StatusOK).
+		Body(strings.NewReader(fixtures.StripeCheckoutSessionWithoutSub))
+	// Set up a response with an inactive subscription.
+	gock.New("https://api.stripe.com").
+		Get("/v1/checkout/sessions/" + sessionIDInactiveSub).
+		Reply(http.StatusOK).
+		Body(strings.NewReader(fixtures.StripeCheckoutSessionWithInactiveSub))
+	// Set up a response without a subscription.
+	gock.New("https://api.stripe.com").
+		Get("/v1/checkout/sessions/" + sessionIDPricelessSub).
+		Reply(http.StatusOK).
+		Body(strings.NewReader(fixtures.StripeCheckoutSessionWithPricelessSub))
+
+	// Get the info on a $20 checkout session.
+	info, status, err := at.StripeCheckoutIDGET(sessionIDWithSub20)
+	if err != nil || status != http.StatusOK {
+		t.Fatal(err, status)
+	}
+	// Ensure the price is correct.
+	if info.Price.ID != priceID20 {
+		t.Fatalf("Expected price '%s', got '%s'", priceID20, info.Price.ID)
+	}
+	// Ensure that the user has been promoted.
+	u, _, err = at.UserGET()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if u.Tier != database.TierPremium20 {
+		t.Fatalf("Expected tier %d, got %d", database.TierPremium20, u.Tier)
+	}
+	// Get the info on a $5 checkout session.
+	info, status, err = at.StripeCheckoutIDGET(sessionIDWithSub5)
+	if err != nil || status != http.StatusOK {
+		t.Fatal(err, status)
+	}
+	// Ensure the price is correct.
+	if info.Price.ID != priceID5 {
+		t.Fatalf("Expected price '%s', got '%s'", priceID5, info.Price.ID)
+	}
+	// Ensure that the user has NOT been demoted.
+	u, _, err = at.UserGET()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if u.Tier != database.TierPremium20 {
+		t.Fatalf("Expected tier %d, got %d", database.TierPremium20, u.Tier)
+	}
+	// Get the info on a checkout session that hasn't been completed and
+	// doesn't have a subscription assigned to it, yet.
+	info, status, err = at.StripeCheckoutIDGET(sessionIDWithoutSub)
+	errStr := "this checkout session does not have an associated subscription"
```

Comment:
I think it would be good to extract these error strings in a constant and then reuse the error constant in the tests, that way the test won't fail if you update the error message later, also it's more readable if it would've been `errNoAssociatedSubscription` or w/e